web/html/src/manager/state/recurring-states.js

```diff
@@ -112,50 +112,43 @@ class RecurringStates extends React.Component {
             "/rhn/manager/api/recurringactions/save",
             JSON.stringify(schedule),
             "application/json"
-        ).promise.then((data) => {
-            let newMsgs = [];
-            const decorator = data.success ? MessagesUtils.info : MessagesUtils.error;
-            if (data.messages === undefined || data.messages.length === 0) {
-                // no explicit messages from the server -> let's display a generic one
-                const defaultMsg = data.success
-                      ? <span>{t("Schedule successfully" + (this.state.action === "create" ? " created." : " updated."))}</span>
-                      : <span>{t("Error on saving schedule.")}</span>;
-                newMsgs = decorator(defaultMsg);
-            }
-            else {
-                // messages from the recurringactions are already localized
-                newMsgs = decorator.apply(null, data.messages);
-            }
-
-            const msgs = this.state.messages.concat(newMsgs);
+        ).promise.then((_) => {
+            const successMsg = <span>{t("Schedule successfully" + (this.state.action === "create" ? " created." : " updated."))}</span>
+            const msgs = this.state.messages.concat(MessagesUtils.info(successMsg));
 
             while (msgs.length > messagesCounterLimit) {
                 msgs.shift();
             }
 
-            this.onMessageChanged(msgs);
+            this.onMessageChanged(msgs); // todo remove - unneeded
             this.setState({
                 messages: msgs
             });
 
-            if (data.success) {
-                this.handleForwardAction();
-            } else {
-                this.handleResponseError();
-            }
-
+            this.handleForwardAction();
         }).catch(this.handleResponseError);
     }
+//
 
     deleteSchedule(item) {
         return Network.del("/rhn/manager/api/recurringactions/" + item.recurringActionId + "/delete")
-            .promise.then(data => {
-                this.handleForwardAction();
+            .promise.then((_) => {
                 this.setState({
                     messages: MessagesUtils.info("Schedule \'" + item.scheduleName + "\' has been deleted.")
                 });
+                this.handleForwardAction();
             })
-            .catch(this.handleResponseError);
+            .catch(data => {
+                const taskoErrorMsg = MessagesUtils.error(
+                    t("Error when scheduling the action. Check if Taskomatic is running"));
```

Comment:
We should put `... when deleting the ...` here instead

Reply:
Good point. Another option would be `... unscheduling ... `. Let me think about it and fix it!